MSVC: escaped backslash in rsp file is processed incorrectly

We have an issue that is somewhat related to #1072, as far as I can tell: we have a preprocessor definition that contains an escaped backslash, which ends up being passed to `cl.exe`as a non-escaped backslash, making our compilation fail. It works correctly when building with `ninja` but not when building with Visual Studio, I assume it is due to the use of `.rsp` files? I made a minimal repro case similar to the one in #1072.

### How to reproduce ###
<!-- Steps to reproduce the behavior: -->

`a.cpp`
```
const char c = DIRSEP;
```
`a.rsp`
```
"-DDIRSEP='\\'"
```
Executing
1. `cl /c a.cpp @a.rsp`
2. `ccache cl /c a.cpp @a.rsp`

### Actual behavior ###
<!-- What happened? -->
1. compiles without error
2. fails compiling with the following error:

> a.cpp(1): error C2001: newline in constant
> a.cpp(2): fatal error C1004: unexpected end-of-file found

because `ccache` passes `-DDIRSEP='\'` to `cl.exe` instead of `-DDIRSEP='\\'`

### Expected behavior ###
<!-- What did you expect to happen? -->
Both compile without error


### Environment ###
<!-- Which ccache version did you use? -->

ccache version 4.6.1, 4.7.3
VS 2019

<!-- If applicable, add more info such as compiler, OS and file system below: -->

--- a/src/Args.cpp
+++ b/src/Args.cpp
@@ -77,7 +77,7 @@ Args::from_atfile(const std::string& filename, AtFileFormat format)
         }
         break;
       case AtFileFormat::msvc:
-        if (*pos != '"' && *pos != '\\') {
+        if (*pos != '"') {
           pos--;
         }
         break;

--- a/unittest/test_Args.cpp
+++ b/unittest/test_Args.cpp
@@ -139,12 +139,12 @@ TEST_CASE("Args::from_atfile")
     CHECK(args[6] == "seve\nth");
   }
 
-  SUBCASE("Only escape double quote and backslash in alternate format")
+  SUBCASE("Only escape double quote in alternate format")
   {
-    util::write_file("at_file", "\"\\\"\\a\\ \\\\\\ \\b\\\"\"\\");
+    util::write_file("at_file", "\"\\\"\\a\\ \\b\\\"\"\\");
     args = *Args::from_atfile("at_file", Args::AtFileFormat::msvc);
     CHECK(args.size() == 1);
-    CHECK(args[0] == "\"\\a\\ \\\\ \\b\"\\");
+    CHECK(args[0] == "\"\\a\\ \\b\"\\");
   }
 
   SUBCASE("Ignore single quote in alternate format")
@@ -154,6 +154,14 @@ TEST_CASE("Args::from_atfile")
     CHECK(args.size() == 2);
     CHECK(args[0] == "'a");
     CHECK(args[1] == "b'");
+  }
+
+  SUBCASE("Do not escape backslash in alternate format")
+  {
+    util::write_file("at_file", "\"-DDIRSEP='\\\\'\"");
+    args = *Args::from_atfile("at_file", Args::AtFileFormat::msvc);
+    CHECK(args.size() == 1);
+    CHECK(args[0] == "-DDIRSEP='\\\\'");
   }
 }
 

